keylime/tenant.py

```diff
@@ -1162,26 +1177,28 @@ def do_add_allowlist(self, args):
         response = cv_client.post(
             f"/v{self.api_version}/allowlists/{self.ima_policy_name}", data=body, timeout=self.request_timeout
         )
-        Tenant._print_json_response(response)
+        Tenant._jsonify_response(response, True)
 
     def do_delete_allowlist(self, name):
         cv_client = RequestsClient(self.verifier_base_url, True, tls_context=self.tls_context)
         response = cv_client.delete(f"/v{self.api_version}/allowlists/{name}", timeout=self.request_timeout)
-        Tenant._print_json_response(response)
+        Tenant._jsonify_response(response, True)
 
     def do_show_allowlist(self, name):  # pylint: disable=unused-argument
         cv_client = RequestsClient(self.verifier_base_url, True, tls_context=self.tls_context)
         response = cv_client.get(f"/v{self.api_version}/allowlists/{name}", timeout=self.request_timeout)
         print(f"Show allowlist command response: {response.status_code}.")
-        Tenant._print_json_response(response)
+        Tenant._jsonify_response(response, True)
 
     @staticmethod
-    def _print_json_response(response):
+    def _jsonify_response(response, print_response=False):
         try:
             json_response = response.json()
         except ValueError:
```

Comment:
The difference to 'before' is that all `response.json()` you converted would have thrown a ValueError on bad input and now you change it to an empty map if there's an error . If you now raised the error in the except clause you would approximately get the same result again as before...

Reply:
Very good point. I am changing the behavior of the function to **optionally** raise an exception, thus keeping the usual behavior for the "other" methods. In the future, I will re-organize and standardize the behavior on all methods.